### DISCUSSION TITLE
Will you ever add a super simple level editor to thextech

### DISCUSSION THREAD

**Discussion by ghost** [Wohlstand/TheXTech#193] [Ideas]
Will you ever add a super simple level editor to thextech (yes i know pge exists) but i feel like it will be nice for thextech to have its own super simple editor

**Comment by Wohlstand** [comment]
Something @ds-sloth is trying to make, and there is a WIP thing presented here

**Comment by ghost** [comment]
So will it be a feature soon?

**Comment by Wohlstand** [comment]
Ye, and mainly it will serve as a small thing for experimenting or using on devices where Moondust Editor won't work. You may ask @ds-sloth for details on how this built-in Editor will work.

The mainstream development toolkit for serious projects the Moondust Devkit is intended (soon I will release the "TheXTech SDK" config pack to allow the use of TheXTech's unique features at some in-game objects, etc.).

P.S. "PGE" is old name, the Moondust Project is now a main title for the project.

**Comment by ghost** [comment]
Okay but the built in editor will still work on pc right? Also thanks for the response

**Comment by Wohlstand** [comment]
Yes, will, right now it's disabled here because it's incomplete, will be enabled once it gets fully implemeted

**Comment by ghost** [comment]
Is their a way to test it right now or do we have to wait for the full release

**Comment by Wohlstand** [comment]
> Is their a way to test it right now or do we have to wait for the full release

Let's @ds-sloth will reply to this question, right now it was used at the WIP 3DS port, I tried to run it locally, but it didn't work, will fix some code on my side.

However, some GIF made by ds-sloth to show how it works on PC:

![Scr_2021-05-28_10-36-51](https://user-images.githubusercontent.com/6751442/130676365-6f0b1f7e-e79a-4b7a-8feb-471272836865.gif)

**Comment by ghost** [comment]
Does @ds-sloth have an early build for the pc i can check out

**Comment by Wohlstand** [comment]
The source code right now at the `master` branch to build, use the CMake flags to enable it, but, you will also need some extra assets files to get it working, lemme upload them...

**Comment by ds-sloth** [comment]
Hi @Derpez! Thanks for your interest.

The editor was feature complete but not polished. @Wohlstand has been making improvements and added 2 way warps which are not fully supported by the editor right now. I haven't been able to develop as much recently since I just moved.

Please follow Wohlstand's instructions. (Thanks for uploading the assets Wohlstand!)

Once I have the time to make the editor feature complete again I will add it to the default build so you just need to set a flag to access it.

**Comment by ghost** [comment]
Ok i will see

**Comment by Wohlstand** [comment]
About assets:
- download the SMBX assets from the official home
- Download this archive: [ui.zip](https://github.com/Wohlstand/TheXTech/files/7041650/ui.zip) and unpack its content into `graphics/ui` directory of SMBX assets you downloaded
- follow the build from sources guide to building the game (you'll find it at wiki)
also, note, it's possibly a CMake flag you need to enable to allow Editor use, but something was changed I missed up (by myself I work on `stable-1.3.5.x` branch to fix up some bugs until I will release the 1.3.5.2 version, then I'll polish some at master branch after)

**Comment by ghost** [comment]
Okay! I will try right now and if i have any more questions or troubles i will come back to this thread

**Comment by ghost** [comment]
Also im not sure where to download the SMBX assets?

**Comment by ghost** [comment]
Like The assets from the SMBX Files?

**Comment by Wohlstand** [comment]
> Like The assets from the SMBX Files?

The assets package for TheXTech, there is a mention at the build manual, you'll find it

i.e. https://github.com/Wohlstand/TheXTech/wiki/Game-assets-packages

**Comment by ghost** [comment]
Thanks!

**Comment by ghost** [comment]
After i build what do i do?

**Comment by ds-sloth** [comment]
Make sure all the assets are available in the game directory or in your pge user directory and then run the game. I believe there should be a menu option "Editor" for you to use.

**Comment by ghost** [comment]
ui folder only has 4 images that are already in graphics/ui

**Comment by ghost** [comment]
it doesnt come with any config or anything

**Comment by ds-sloth** [comment]
If they're already there then don't worry about it. Is "Editor" available in the main menu?

**Comment by ghost** [comment]
no

**Comment by ghost** [comment]
the version of the game i downloaded im pretty sure was the master release so i dont know

**Comment by ghost** [comment]
but basically i took a built version of the game from my release and put it in assets folder and put the ui in graphics/ui but no editor in menu

**Comment by ghost** [comment]
i think it might require some special config for it to work

**Comment by ds-sloth** [comment]
Okay, @Wohlstand @Derpez, it actually looks like I did set this up as a runtime config flag already. Sorry for not remembering. You'll want to edit settings.ini to enable the new editor. You probably need to replace a zero with a one if I remember right.

**Comment by ghost** [comment]
okay ill try

**Comment by ghost** [comment]
@ds-sloth their is no settings.ini in both of my builds

**Comment by ghost** [comment]
i opened all ini's and couldn't find editor

**Comment by ds-sloth** [comment]
This depends on whether you have set your settings up to be portable or not. Your settings are probably in your home directory under PGE project.

**Comment by ghost** [comment]
nope
![Capture](https://user-images.githubusercontent.com/85128296/130686678-126fa8ba-6a45-4d84-8cc3-ca4f25d75006.PNG)

**Comment by ds-sloth** [comment]
Try looking directly in .PGE_Project/thextech

**Comment by ghost** [comment]
nothing

**Comment by Wohlstand** [comment]
Lol, you should put the entire content of THIS directory on screenshot into ".PGE_Project/thextech"

**Comment by ghost** [comment]
wdym

**Comment by Wohlstand** [comment]
> wdym

what means?

**Comment by ghost** [comment]
what do you mean

**Comment by Wohlstand** [comment]
Ah, move ALL files and sub-directores from `.PGE_Project/thextech/thextech-smbx13-assets-full/` directory into `.PGE_Project/thextech/`

**Comment by ghost** [comment]
ah okay but still i cant find editor in any of ini files

**Comment by ghost** [comment]
@ds-sloth any ideas

**Comment by Wohlstand** [comment]
The question: which executable build you had to download? If you want the thing in debug mode (this way), you should open the "Fresh devel builds" page and find the plain executable packages
https://github.com/Wohlstand/TheXTech/wiki/Fresh-devel-builds

but, in some cases, you should build the entire game yourself from sources:
(especially the way to build for Windows: https://github.com/Wohlstand/TheXTech/wiki/Building-on-Windows)

**Comment by Wohlstand** [comment]
All INI files got automatically created at the `.PGE_Project/thextech/settings` sub-directory (it will be auto-created if not exist)

**Comment by ghost** [comment]
i tried to build with mingw from cmd but it failed 

== TheXTech version 1.3.6 ==
CMake Error at CMakeLists.txt:158 (add_subdirectory):
  The source directory

    C:/Users/******/Desktop/xtech dev/TheXTech/3rdparty/PGE_File_Formats

  does not contain a CMakeLists.txt file.


-- Logger for Desktop
-- Performing Test TCLAP_HAVE_STRSTREAM
-- Performing Test TCLAP_HAVE_STRSTREAM - Failed
--- Detected system libraries list: version;opengl32;dbghelp;advapi32;kernel32;winmm;imm32;setupapi ---
MinGW DLLs: [C:/mingw64/bin/libgcc_s_seh-1.dll;C:/mingw64/bin/libwinpthread-1.dll;C:/mingw64/bin/libstdc++-6.dll]
-- Configuring incomplete, errors occurred!
See also "C:/Users/******/Desktop/xtech dev/TheXTech/build/CMakeFiles/CMakeOutput.log".
See also "C:/Users/******/Desktop/xtech dev/TheXTech/build/CMakeFiles/CMakeError.log".

**Comment by Wohlstand** [comment]
You forgot to download submodules, lol, please use the git to clone the repository, don't download just the zip archive, it will be broken. Mainly because it doesn't contain required submodules inside. The manual tells about that (how to clone the repository and how to pull submodules, see "Intro" paragraph).

**Comment by ghost** [comment]
oh forgot to do --recursive

**Comment by Wohlstand** [comment]
No worry, just run:
```
git submodule init
git submodule update
```

**Comment by ds-sloth** [comment]
You shouldn't need to do your own build. Current builds will support this flag fine. I'm sorry for not remembering sooner.

**Comment by ghost** [comment]
its okay i just cant find editor flag any where

**Comment by Wohlstand** [comment]
Seems it was been removed, and the master build contains all things by default :thinking:  (I checked the stuff just now)

**Comment by ghost** [comment]
the build still failed same error

**Comment by ghost** [comment]
yea i tried going into the ini files and adding editor=0 or something and didnt work

**Comment by ghost** [comment]
editor=1 my bad

**Comment by ghost** [comment]
@ds-sloth can you acces the editor right now?

**Comment by ds-sloth** [comment]
The flag should be `new-editor=1`, not on by default, in the `main` section.
